R/evaluate_test_code.R

```diff
@@ -0,0 +1,80 @@
+#' Evaluate the test code file
+#' @param file. full path to test code file.
+#' @param test_env environment to perform tests in
+#' @param ... argument passed to `knitr::kable()`
+#' @param ref reference path to use. Defaults to vt_path()
+#' @return a kable with variables: \code{Test}, \code{Expected}, \code{Results},
+#' \code{Pass/Fail}. Suitable for including in validation report
+#' @export
+#' @importFrom testthat capture_output Reporter
+#' @importFrom knitr kable
+vt_run_test_code_file <- function(file, test_env  = new.env(), ..., ref = vt_path()){
+
+  test_results <- eval_test_code(path = file.path(ref, "test_code",file), test_env = test_env)
+
+  return(kable(test_results, ...))
+}
+
+#' @importFrom testthat capture_output Reporter test_file
+#' @importFrom rlang warn
+#' @noRd
+eval_test_code <- function(path, test_env = new.env()) {
+
+  # Get the testthat results (via reporter)
+  # Used for obtaining the Pass/Fail variable
+  output <- testthat::test_file(path , reporter = testthat::Reporter,env = test_env)
```

Comment:
Lets open an issue to track that. I think it would be good for a future release. Functionally all we care about is grabbing the outcomes of each expectation.